presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/MapColumnStatisticsBuilder.java

```diff
@@ -25,14 +25,22 @@
 import java.util.Optional;
 
 import static com.facebook.presto.orc.metadata.statistics.ColumnStatistics.mergeColumnStatistics;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
 import static java.util.Objects.requireNonNull;
 
 public class MapColumnStatisticsBuilder
         implements StatisticsBuilder
 {
+    private final ImmutableList.Builder<MapStatisticsEntry> entries = new ImmutableList.Builder<>();
+    private final boolean collectKeyStats;
     private long nonNullValueCount;
     private boolean hasEntries;
-    private final ImmutableList.Builder<MapStatisticsEntry> entries = new ImmutableList.Builder<>();
+
+    public MapColumnStatisticsBuilder(boolean collectKeyStats)
```

Comment:
This new argument simplifies usage of the statistics builder for both cases with enabled and disabled map stats.